src/github-web-worker.js

```diff
@@ -1,257 +1,260 @@
-                }
-
-                resolve({
-                    type: 'success',
-                    message: 'Retrieved DB from <strong>GitHub</strong> successfully',
-                    db: JSON.parse(data.files['project-viewer.json'].content),
-                    options: {
-                        icon: 'mark-github'
-                    }
-                });
-            });
-        });
-        return promise;
-    },
-    checkForGist: function checkForGist() {
-        const promise = new Promise((resolve, reject) => {
-
-            let headers = new Headers();
-
-            if (!this.token) {
-                reject({
-                    type: 'warning',
-                    message: 'no <strong>Github</strong>\'s token was provided, please check the configuration.',
-                    options: {
-                        icon: 'mark-github'
-                    }
-                });
-                return;
-            }
-
-            headers.append('Accept', 'application/vnd.github.v3+json');
-            headers.append('Authorization', 'token ' + this.token);
-
-            let parameters = {
-                method: 'GET',
-                headers: headers
-            };
-
-            fetch(this.url, parameters)
-            .then(function(response) {
-                return response.json();
-            })
-            .then((data) => {
-                if (!data || data.length === 0) {
-                    this.setPrivateGist().then(resolve).catch(reject);
-                } else {
-                    let hasGist = data.some((gist) => {
-                        let validation = gist.files.hasOwnProperty('project-viewer.json');
-                        if (validation) {
-                            this.gistId = gist.id;
-                        }
-                        return validation;
-                    });
-
-                    if (hasGist) {
-                        this.getPrivateGist().then(resolve).catch(reject);
-                    } else {
-                        this.setPrivateGist().then(resolve).catch(reject);
-                    }
-                }
-            });
-        });
-        return promise;
-    }
-};
-
-onmessage = function(e) {
-    if (!e.data || e.data.length === 0) {
-        return;
-    }
-
-    if (e.data[0].hasOwnProperty('token')) {
-        api.token = e.data[0].token;
-    }
-
-    if (e.data[0].action === 'fetch') {
-        api.checkForGist()
-        .then(postMessage)
-        .catch(postMessage);
-    }
-    else if (e.data[0].action === 'update') {
-        api.checkForGist()
-        .then(() => {
-            api.updatePrivateGist(e.data[0].value)
-            .then(postMessage)
-            .catch(postMessage);
-        })
-        .catch(postMessage);
-    }
-
-
-}
+const setNamePlaceholder = '{set-name}';
+const api = {
+  token: undefined,
+  gistId: undefined,
+  setName: 'default',
+  gistFilePattern: 'project-viewer-' + setNamePlaceholder + '.json',

```

Comment:
let's abuse ES6, change it to ``gistFilePattern: `project-viewer-{$setNamePlaceholder}.json`,``